ably/ably-chat-js
# dev: additional eslint rules

### Description

- Adds new vitest expectations for synchronously throwing `ErrorInfo` - avoids us having to do things async just to assert on `ErrorInfo`.
- Enables the recommended config for eslint-plugin-unicorn. This enforces some additional rules for consistency and readability of code such as:
  - Not returning `Promise.resolve()` for the sake of it
  - Not abbreviating error to err
  - Using braces to clearly delimit switch status
  - Avoiding `Array.forEach` in favour of `for..of`
  - Using `undefined` rather than `null` where possible

### Checklist


### Testing Instructions (Optional)

- Check ESLint passes
- Check tests pass
- Try using the demo app
